bubble_sort.rb

```diff
@@ -0,0 +1,37 @@
+def bubble(array)
+  len_ = array.length - 1
+  loop do
+    sorted = false
+    len_.times do |x|
+      if array[x] > array[x + 1]
+        array[x], array[x + 1] = array[x + 1], array[x]
+        sorted = true
+      end
+    end
+    len_ -= 1
+    break unless sorted
+  end
+  array
+end
+
+def bubble_sort_by(array)
+  len_ = array.length - 1
+  loop do
+    sorted = false
+    len_.times do |x|
+      if yield(array[x], array[x + 1]).positive?
+        array[x], array[x + 1] = array[x + 1], array[x]
+        sorted = true
+      end
+    end
+    len_ -= 1
+    break unless sorted
+  end
+  print array
```

Comment:
Your method should return the sorted array and not print it. As it is now, it returns nil.